app/lib/features/space/sheets/link_chat_sheet.dart

```diff
@@ -0,0 +1,224 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/widgets/default_button.dart';
+import 'package:acter/common/widgets/search.dart';
+import 'package:acter/common/widgets/side_sheet.dart';
+import 'package:acter/common/widgets/spaces/select_space_form_field.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class LinkChatPage extends ConsumerStatefulWidget {
+  final String? parentSpaceId;
+
+  const LinkChatPage({super.key, required this.parentSpaceId});
+
+  @override
+  ConsumerState<LinkChatPage> createState() => _LinkChatPageConsumerState();
+}
+
+class _LinkChatPageConsumerState extends ConsumerState<LinkChatPage> {
+  final TextEditingController searchTextEditingController =
+      TextEditingController();
+  final List<String> subChatsIds = [];
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      selectParentSpaceData();
+    });
+    super.initState();
+  }
+
+  //Select parent space data
+  void selectParentSpaceData() {
+    if (widget.parentSpaceId != null) {
+      final notifier = ref.read(selectedSpaceIdProvider.notifier);
+      notifier.state = widget.parentSpaceId;
+    }
+  }
+
+  //Fetch known sub-chats list of selected parent space
+  void fetchKnownSubChatsData() {
+    final selectedParentSpaceId = ref.watch(selectedSpaceIdProvider);
+    if (selectedParentSpaceId == null) return;
+    final sp = ref.watch(spaceRelationsOverviewProvider(selectedParentSpaceId));
+    sp.when(
+      data: (space) {
+        subChatsIds.clear();
+        for (int i = 0; i < space.knownChats.length; i++) {
+          subChatsIds.add(space.knownChats[i].getRoomId().toString());
+        }
+      },
+      error: (e, s) => Container(),
+      loading: () => Container(),
```

Comment:
use [`whenData`](https://pub.dev/documentation/hooks_riverpod/latest/hooks_riverpod/AsyncValueX/whenData.html) when only handling the data case?
```suggestion
    sp.whenData( (space) {
        subChatsIds.clear();
        for (int i = 0; i < space.knownChats.length; i++) {
          subChatsIds.add(space.knownChats[i].getRoomId().toString());
        }
      },);
```